### ISSUE TITLE
pg_dump fails

### ISSUE THREAD

**Issue by dtaelman** [pgpointcloud/pointcloud#197]
Hi all,

Performing a backup of my postgreSQL databases (using PostgreSQL 10.3) with the pointcloud-extension enabled fails and generates the following errors:
- pg_dump: Dumping the contents of table "points" failed: PQgetResult() failed.
- pg_dump: Error message from server: ERROR:  relation "pointcloud_formats" does not exist
- LINE 1: select schema, srid from pointcloud_formats where pcid = 1
- QUERY:  select schema, srid from pointcloud_formats where pcid = 1
- pg_dump: The command was: COPY public.points (id, pt) TO stdout

Does anyone have an idea what causes the pg_dump to fail and how to solve the issue?

**Comment by elemoine** [comment]
Can you please provide the full pg_dump command line that you're trying to use? Thanks.

**Comment by dtaelman** [comment]
The full pg_dump command is:
pg_dump --file "/Users/devitaelman/Desktop/test.sql" --host "localhost" --port "5432" --username "devitaelman" --verbose --format=p “lidartest”

Thanks
Devi

**Comment by trey-stafford** [comment]
I have also experienced this issue with postgresql 9.6.8, pointcloud v1.1.0:

```
# pg_dump -U $DB_NAME -U $DB_USER -F p --file /share/appdata/dump/db.dump

g_dump: Dumping the contents of table "atm1b" failed: PQgetResult() failed.
pg_dump: Error message from server: ERROR:  relation "pointcloud_formats" does not exist
LINE 1: select schema, srid from pointcloud_formats where pcid = 1
                                                     ^
QUERY:  select schema, srid from pointcloud_formats where pcid = 1
pg_dump: The command was: COPY public.atm1b (id, pa) TO stdout
```

**Comment by trey-stafford** [comment]
Note that I _can_ dump the pointcloud_formats table:
```
pg_dump -U $DB_USER -F p --file /share/appdata/dump/db.dump -t pointcloud_formats
```

**Comment by gabri94** [comment]
I have the very same error trying to upgrade from postgres 9.5 to postgres 9.6
see #211

**Comment by trey-stafford** [comment]
I have come back to this issue again recently, and have tested postgresql versions v9.6.7 through v9.6.10.

The problem with pg_dump appears in postgresql v9.6.8 through v9.6.10.

pg_dump does work with postgresql v9.6.7.

**Comment by foggel** [comment]
I have the same problem and wrote a patch for pc_schema_from_pcid_uncached in psql/pc_pgsql.c to prefix pointcloud_formats with the schema name where the extension is installed. The schema name can be found by querying pg_catalog.pg_extension.

[issue_197.txt](https://github.com/pgpointcloud/pointcloud/files/2749155/issue_197.txt)

I don't have much experience in c-programming or postgres extensions so it might be safer to just prefix the query with "public" until someone more experienced fixes this.

**Comment by autra** [comment]
Reproducing this too. It's definitely a search_path issue. Prefixing `POINTCLOUD_FORMATS` variable with `public` does fix the generation of dumps (but not the restore). Patch linked by @foggel fixes it all for me. Something I find odd: why do we need to read the schema to dump? 

@foggel do you plan on opening a PR with your fix?

**Comment by foggel** [comment]
> @foggel do you plan on opening a PR with your fix?
I will try to find som time the next days to create a PR.

**Comment by anibalrivero** [comment]
Hi! I am experiencing this problem. If I get this right, If I clone the master branch this should be solved, or how can I use version 1.3.0?

**Comment by autra** [comment]
Hi @anibalrivero. Yes exactly. As far as I know, v1.3.0 hasn't yet been packaged in pgdg, so your only option right now is to compile it.
